### PR TITLE
Enables configuration for flux version and container repository

### LINKED ISSUE #284: Enable operations to pin Flux version
As an operator, I don't want a regression or functionality change in Flux to potentially compromise my ability to do a deployment at any time.  To support this, we should add the ability to pin a particular release of Flux such that they can stay on it for production clusters until they have validated newer versions on non-production clusters.

### PR DESCRIPTION
**Pull request by sarath-p**
Allows configuration for flux source, container registry for container and helm chart download.

Closes #284 and #343

### CODE REVIEW

File: cluster/common/flux/variables.tf [hunk 1]
```diff
@@ -4,6 +4,18 @@ variable "flux_repo_url" {
   default = "https://github.com/weaveworks/flux.git"
 }
 
+# container registry to download flux image
+variable "flux_image_repository" {
+  type = "string"
+  default = "docker.io/weaveworks/flux"
```
Comment (jmspring) [inline cluster/common/flux/variables.tf]: You have the default here, yet it needs to be passed in via aks-gitops?

File: cluster/common/flux/variables.tf [hunk 2]
```diff
@@ -4,6 +4,18 @@ variable "flux_repo_url" {
...
+}
+
+# flux version to download source from git repo and container image from the registry
+variable "flux_image_tag" {
+  type = "string"
+  default = "1.12.1"
```
Comment (jmspring) [inline cluster/common/flux/variables.tf]: As above, default here, but passed in via aks-gitops?

File: cluster/environments/azure-simple/variables.tf [hunk 3]
```diff
@@ -27,6 +27,18 @@ variable "flux_recreate" {
   default     = ""
 }
 
+# container registry to download flux image
+variable "flux_image_repository" {
+  type = "string"
```
Comment (jmspring) [inline cluster/environments/azure-simple/variables.tf]: Default values should be isolated to one location

File: cluster/environments/azure-simple/variables.tf [hunk 4]
```diff
@@ -27,6 +27,18 @@ variable "flux_recreate" {
...
+  default = "docker.io/weaveworks/flux"
+}
+
+# flux version to download source from git repo and container image from the registry
+variable "flux_image_tag" {
+  type = "string"
+  default = "1.12.1"
```
Comment (jmspring) [inline cluster/environments/azure-simple/variables.tf]: As above

File: cluster/environments/azure-single-keyvault/variables.tf [hunk 5]
```diff
@@ -29,6 +29,18 @@ variable "flux_recreate" {
   default     = ""
 }
 
+# container registry to download flux image 
+variable "flux_image_repository" {
+  type    = "string"
```
Comment (jmspring) [inline cluster/environments/azure-single-keyvault/variables.tf]: As above

File: cluster/environments/azure-single-keyvault/variables.tf [hunk 6]
```diff
@@ -29,6 +29,18 @@ variable "flux_recreate" {
...
+  default = "docker.io/weaveworks/flux"
+}
+
+# flux version to download source from git repo and container image from the registry
+variable "flux_image_tag" {
+  type    = "string"
```
Comment (jmspring) [inline cluster/environments/azure-single-keyvault/variables.tf]: As above

File: cluster/common/flux/variables.tf (hunk 1 above)
Comment (sarath-p) [reply cluster/common/flux/variables.tf]: Yes. The scenario is flux module has the default value, but wanted to override the it only in `aks-gitops` implementation.

File: cluster/environments/azure-simple/variables.tf (hunk 3 above)
Comment (sarath-p) [reply cluster/environments/azure-simple/variables.tf]: This PR provides not only implementing the default values in the module, but also extending it to override those default values in environment templates.

**Pull request merged**